scipy/optimize/_direct_py.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+r"""
+scipydirect - A python wrapper to the DIRECT algorithm.
+=======================================================
+
+DIRECT is a method to solve global bound constraint optimization problems and
+was originally developed by D. R. Jones, C. D. Perttunen and B. E. Stuckmann.
+It is designed to find **global** solutions of mathematical optimization problems of the from
+
+.. math::
+
+       \min_ {x \in R^n} f(x)
+
+subject to
+
+.. math::
+
+       x_L \leq  x  \leq x_U
+
+Where :math:`x` are the optimization variables (with upper and lower
+bounds), :math:`f(x)` is the objective function.
+
+The DIRECT package uses the Fortran implementation of DIRECT written by
+Joerg.M.Gablonsky, DIRECT Version 2.0.4. More information on the DIRECT
+algorithm can be found in Gablonsky's `thesis <http://repository.lib.ncsu.edu/ir/bitstream/1840.16/3920/1/etd.pdf>`_.
+
+.. codeauthor:: Andreas Mayer <andimscience@gmail.com>, Amit Aides <amitibo@tx.technion.ac.il>
+"""
+
+from __future__ import print_function
+import numpy as np
+from ._direct import direct
+from .optimize import OptimizeResult
+
+ERROR_MESSAGES = (
+    'u[i] < l[i] for some i',
+    'maxfun is too large',
+    'Initialization failed',
+    'There was an error in the creation of the sample points',
+    'An error occured while the function was sampled',
+    'Maximum number of levels has been reached.',
+)
+
+SUCCESS_MESSAGES = (
+    'Number of function evaluations done is larger then maxfun',
```

Comment:
I think `SUCCESS_MESSAGES` might mean that, the algorithm terminated successfully though, number of function evaluations were larger than `maxfun`. Does, `FINAL_MESSAGES` sound good as a variable name?

Reply:
> The  program  termination  criteria  differs  depending  on  the  value  of  the opts(5).   If the value is set to 1, DIRECT will terminate as soon as it exceeds its budget of iterations,rectangle divisions, or function evaluations.  The function evaluation budget is a tentative value, since DIRECT may exceed this value by a slight amount if it is in the middle of an iteration when the budget has been exhausted.

The above quoted from page 3 of [here](http://www2.peq.coppe.ufrj.br/Pessoal/Professores/Arge/COQ897/Naturais/DirectUserGuide.pdf). It seems like the these messages are success messages because, number of function evaluations may exceed but algorithm will still succeed. The values are not restrictive but are used for stopping criteria when global minimum is not known.